sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations/IVirtualMachinesOperations.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Microsoft.Azure.Management.Compute
+{
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    /// VirtualMachinesOperations operations.
+    /// </summary>
+    public partial interface IVirtualMachinesOperations
+    {
+        /// <summary>
+        /// The operation to delete a virtual machine.
+        /// </summary>
+        /// <param name='resourceGroupName'>
+        /// The name of the resource group.
+        /// </param>
+        /// <param name='vmName'>
+        /// The name of the virtual machine.
+        /// <param name='customHeaders'>
+        /// The headers that will be added to request.
+        /// </param>
+        /// <param name='cancellationToken'>
+        /// The cancellation token.
+        /// </param>
+        /// <exception cref="Microsoft.Rest.Azure.CloudException">
+        /// Thrown when the operation returned an invalid status code
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null
+        /// </exception>
+        Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string vmName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
So, this isn't quite right.  We have to be explicit about the existing overloads, rather than use default parameter values, so instead, the custom interface and implementation should have two new methods instead of one:

``` c#
Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string vmName, Dictionary<string, List<string>> customHeaders);
Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string vmName, Dictionary<string, List<string>> customHeaders, CancellationToken cancellationToken);
```

This pattern should be followed throughout.  Without this, there are a few combinations 
like ```DeleteWithHttpMessagesAsync("rg", "vm")```,  ```DeleteWithHttpMessagesAsync("rg", "vm", null)``` that cannot be disambiguated at runtime.

Reply:
Added as suggested.